From a1ad35c90565ea700b0ce6045d0d4b02707a315c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Robert=20L=C3=B6nnqvist?= <robert.lonnqvist@cygni.se>
Date: Mon, 24 Jun 2019 16:39:38 +0200
Subject: [PATCH] Fix term colors for ubuntu

---
 bashrc | 15 ++++++++++-----
 zshrc  | 19 +++++++++++++------
 2 files changed, 23 insertions(+), 11 deletions(-)

diff --git a/bashrc b/bashrc
index 7447476..b3c5716 100644
--- a/bashrc
+++ b/bashrc
@@ -74,14 +74,18 @@ fi
 
 # if no git prompt has been loaded, load one.
 if ! declare -f __git_ps1 >/dev/null 2>&1 ; then
-  if [[ -f /usr/share/git/completion/git-prompt.sh ]]; then
+  if [[ -e /usr/local/etc/bash_completion.d/git-prompt.sh ]]; then
+    . /usr/local/etc/bash_completion.d/git-prompt.sh
+  elif [[ -e /usr/share/git/completion/git-prompt.sh ]]; then
     . /usr/share/git/completion/git-prompt.sh
-  elif [[ -f /usr/share/git-core/contrib/completion/git-prompt.sh ]]; then
+  elif [[ -e /usr/share/git-core/contrib/completion/git-prompt.sh ]]; then
     . /usr/share/git-core/contrib/completion/git-prompt.sh
+  elif [[ -e /usr/lib/git-core/git-sh-prompt ]]; then
+    . /usr/lib/git-core/git-sh-prompt
   fi
 fi
 
-PS1='\[$(__exit_status_color)\]❯ \[\e[01;36m\]\W\[\e[00m\] '
+PS1='\[$(__exit_status_color)\]> \[\e[01;36m\]\W\[\e[00m\] '
 if declare -f __git_ps1 >/dev/null 2>&1 ; then
   # prompt setup
   GIT_PS1_SHOWDIRTYSTATE=1
@@ -89,7 +93,7 @@ if declare -f __git_ps1 >/dev/null 2>&1 ; then
   GIT_PS1_SHOWSTASHSTATE=1
   GIT_PS1_SHOWUNTRACKEDFILES=1
   GIT_PS1_SHOWCOLORHINTS=1
-  PROMPT_COMMAND='__git_ps1 "\[$(__exit_status_color)\]❯ \[\e[01;36m\]\W\[\e[00m\]" " " " (%s)"'
+  PROMPT_COMMAND='__git_ps1 "\[$(__exit_status_color)\]> \[\e[01;36m\]\W\[\e[00m\]" " " " (%s)"'
 fi
 
 # if keychain is installed use it
@@ -109,13 +113,14 @@ alias zfgrep="zfgrep --color=auto"
 
 if [[ -f ~/.dir_colors ]]; then
   eval "$(dircolors -b ~/.dir_colors)"
+else
+  export LS_COLORS="di=1;36:ln=35:so=32:pi=33:ex=31:bd=34;46:cd=34;43:su=30;41:sg=30;46:tw=30;42:ow=30;43"
 fi
 
 # platform specific stuff
 if [[ "$(uname)" == "Darwin" ]]; then
   export CLICOLOR=1
   export LSCOLORS="Gxfxcxdxbxegedabagacad"
-  export LS_COLORS="di=1;36:ln=35:so=32:pi=33:ex=31:bd=34;46:cd=34;43:su=30;41:sg=30;46:tw=30;42:ow=30;43"
   alias ls="ls -GFh"
 else
   alias ls="ls --color=auto -Fh"
diff --git a/zshrc b/zshrc
index 1222bda..5a60767 100644
--- a/zshrc
+++ b/zshrc
@@ -44,7 +44,7 @@ zle -N edit-command-line
 bindkey '^X^E' edit-command-line
 
 # paths
-typeset -U path
+declare -U path
 if [[ -d "${HOME}/.bin" ]]; then
   path=("${HOME}/.bin" $path[@])
 fi
@@ -76,12 +76,17 @@ fi
 if [[ "$(uname)" == "Darwin" ]]; then
   export CLICOLOR=1
   export LSCOLORS="Gxfxcxdxbxegedabagacad"
-  export LS_COLORS="di=1;36:ln=35:so=32:pi=33:ex=31:bd=34;46:cd=34;43:su=30;41:sg=30;46:tw=30;42:ow=30;43"
   alias ls="ls -GFh"
 else
   alias ls="ls --color=auto -Fh"
 fi
 
+if [[ -f ~/.dir_colors ]]; then
+  eval "$(dircolors -b ~/.dir_colors)"
+else
+  export LS_COLORS="di=1;36:ln=35:so=32:pi=33:ex=31:bd=34;46:cd=34;43:su=30;41:sg=30;46:tw=30;42:ow=30;43"
+fi
+
 # functions
 man() {
   env LESS_TERMCAP_mb=$'\e[01;33m' \
@@ -113,13 +118,15 @@ autoload -Uz colors && colors
 # prompt
 setopt promptsubst
 
-if ! typeset -f __git_ps1 2>&1 >/dev/null ; then
+if ! declare -f __git_ps1 2>&1 >/dev/null ; then
   if [[ -e /usr/local/etc/bash_completion.d/git-prompt.sh ]]; then
     . /usr/local/etc/bash_completion.d/git-prompt.sh
   elif [[ -e /usr/share/git/completion/git-prompt.sh ]]; then
     . /usr/share/git/completion/git-prompt.sh
   elif [[ -e /usr/share/git-core/contrib/completion/git-prompt.sh ]]; then
     . /usr/share/git-core/contrib/completion/git-prompt.sh
+  elif [[ -e /usr/lib/git-core/git-sh-prompt ]]; then
+    . /usr/lib/git-core/git-sh-prompt
   fi
 fi
 
@@ -129,10 +136,10 @@ GIT_PS1_SHOWSTASHSTATE=1
 GIT_PS1_SHOWUNTRACKEDFILES=1
 GIT_PS1_SHOWCOLORHINTS=1
 
-PROMPT='%(?:%{$fg_bold[green]%}❯ :%{$fg_bold[red]%}❯ %s)%{$fg[cyan]%}%c%{$reset_color%}%b '
-if typeset __git_ps1 2>&1 >/dev/null ; then
+PROMPT='%(?:%{$fg_bold[green]%}> :%{$fg_bold[red]%}> %s)%{$fg[cyan]%}%c%{$reset_color%}%b '
+if declare -f __git_ps1 2>&1 >/dev/null ; then
   precmd() {
-    __git_ps1 '%(?:%{$fg_bold[green]%}❯ :%{$fg_bold[red]%}❯ %s)%{$fg_bold[cyan]%}%c%b' '%{$reset_color%} ' ' (%s)'
+    __git_ps1 '%(?:%{$fg_bold[green]%}> :%{$fg_bold[red]%}> %s)%{$fg_bold[cyan]%}%c%b' '%{$reset_color%} ' ' (%s)'
   }
 fi
 
-- 
2.20.1

